### ISSUE TITLE
Typographer changes apostrophe after link to left single quote

### ISSUE THREAD

**Issue by rgov** [yuin/goldmark#129]
- [x] goldmark is fully compliant with the CommonMark. Before submitting issue, you **must** read [CommonMark spec](https://spec.commonmark.org/0.29/) and confirm your output is different from [CommonMark online demo](https://spec.commonmark.org/dingus/).
    - [x] **Extensions(Autolink without `<` `>`, Table, etc) are not part of CommonMark spec.** You should confirm your output is different from other official renderers correspond with an extension.
- [x] **goldmark is not dedicated for Hugo**. If you are Hugo user and your issue was raised by your experience in Hugo, **you should consider create issue at Hugo repository at first** .
- [x] Before you make a feature request, **you should consider implement the new feature as an extension by yourself** . To keep goldmark itself simple, most new features should be implemented as an extension.

Please answer the following before submitting your issue:

1. What version of goldmark are you using? : Unclear; came with Hugo master (0.71 beta).
2. What version of Go are you using? : `go version go1.14.2 darwin/amd64`
3. What operating system and processor architecture are you using? : macOS 10.15.4, x86_64
4. What did you do? :

With the typographer extension, I wrote a link with an apostrophe after it, like this:

```
[Cat][]'s Pajamas
```

5. What did you expect to see? :

A right single quote:

```
<a ...>Cat<a>&rsquo;s Pajamas
```

6. What did you see instead? :

A left single quote:

```
<a ...>Cat<a>&lsquo;s Pajamas
```

7. Did you confirm your output is different from [CommonMark online demo](https://spec.commonmark.org/dingus/) or other official renderer correspond with an extension?:

```
$ echo "<a>Cat</a>'s Pajamas" | perl SmartyPants_1.5.1/SmartyPants.pl 
<a>Cat</a>&#8217;s Pajamas
```

`&#8127;` is `&rsquo;`.

**Comment by rgov** [comment]
The rule to follow here is that if the HTML / Markdown markup (whichever you operate on) does not break up a word, it should proceed as if the markup was not there.

E.g., `<a>`, `<span>`, `<b>`, `<i>`, etc. should be ignored for the purpose of deciding quote direction.

**Comment by yuin** [comment]
This is a limitation of the current algorithm. see [#39 (comment)](https://github.com/yuin/goldmark/issues/39#issuecomment-561044672)

As I wrote there, I welcome PR that enhances typographer(or adds complete SmartyPants extension).

**Comment by rgov** [comment]
Thanks. I recognize your prerogative to handle issues as you see fit, but because all of these issues are closed, it is hard for people who are unfamiliar with the project to find this limitation. I would encourage you to consider keeping at least one bug open (probably #39) related to Typographer.

I'd also recommend removing the extension or perhaps at least documenting that it has numerous problems. Due to the current bugs it makes processed text incorrect which could be embarrassing.

**Comment by yuin** [comment]
I've tried to fix this in 3393022.
Except extreme cases, it seems working correctly for me.
